app/src/main/java/net/squanchy/schedule/view/ScheduleDayPageView.kt

```diff
@@ -52,3 +57,19 @@ class ScheduleDayPageView @JvmOverloads constructor(
         override fun areContentsTheSame(oldItemPosition: Int, newItemPosition: Int) = oldEvents[oldItemPosition] == newEvents[newItemPosition]
     }
 }
+
+private class SnappingLinearLayoutManager(context: Context) : LinearLayoutManager(context) {
+
+    override fun smoothScrollToPosition(recyclerView: RecyclerView, state: RecyclerView.State?, position: Int) {
+        val smoothScroller = TopSnappedSmoothScroller(this, recyclerView.context)
+        smoothScroller.targetPosition = position
+        startSmoothScroll(smoothScroller)
+    }
+}
+
+private class TopSnappedSmoothScroller(private val layoutManager: SnappingLinearLayoutManager, context: Context) : LinearSmoothScroller(context) {
+    override fun computeScrollVectorForPosition(targetPosition: Int): PointF? =
```

Comment:
We don't really need this, every built-in LayoutManager in RecyclerView implements `ScrollVectorProvider`, so the SmoothScroller will try to dispatch the call to the LayoutManager attached to it, no need to override it